anti-analysis/anti-debugging/debugger-evasion/hide-thread-from-debugger.yml

```diff
@@ -18,12 +20,19 @@ rule:
       - 26beba7352a32b803aa19e0782011a383a1df19549910e7b2f2f244e49678524:0x10001670
   features:
     - or:
-      - basic block:
-        - and:
-          - or:
-            - api: NtSetInformationThread
-            - api: ZwSetInformationThread
-          - number: 0x11 = ThreadHideFromDebugger
+      - or:
+        - basic block:
+          - and:
+            - or:
+              - api: NtSetInformationThread
+              - api: ZwSetInformationThread
+            - number: 0x11 = ThreadHideFromDebugger
+        - thread:
+          - and:
+            - or:
+              - api: NtSetInformationThread
+              - api: ZwSetInformationThread
+            - number: 0x11 = ThreadHideFromDebugger
```

Comment:
For future improvements, we may think about unifying this to make it easier to read, like `- basic block or thread` maybe?

The real fix is to support call scope, of course

Reply:
yeah, this repetition is really tedious.

it makes the logic hard to follow, and also makes future updates more difficult.

i think we can pretty quickly add some special cases to the rules parser to enable a combination of hard-coded scope combinations. we should remove this once the call scope is implemented though.